xbmc/cores/VideoPlayer/DVDInputStreams/InputStreamAddon.cpp

```diff
@@ -20,40 +20,143 @@
+{
+  // check if a specific inputstream addon is requested
+  CVariant addon = fileitem.GetProperty("inputstreamaddon");
+  if (!addon.isNull())
+  {
+    if (addon.asString() != addonBase->ID())
+      return false;
+    else
+      return true;
+  }
+
+  // check protocols
+  std::string protocol = fileitem.GetURL().GetProtocol();
+  if (!protocol.empty())
+  {
+    std::string protocols = addonBase->Type(ADDON_INPUTSTREAM)->GetValue("@protocols").asString();
+    if (!protocols.empty())
+    {
+      std::vector<std::string> protocolsList = StringUtils::Tokenize(protocols, "|");
+      for (auto& value : protocolsList)
+      {
+        StringUtils::Trim(value);
+        if (value == protocol)
+          return true;
+      }
+    }
+  }
+
+  std::string filetype = fileitem.GetURL().GetFileType();
+  if (!filetype.empty())
+  {
+    std::string extensions = addonBase->Type(ADDON_INPUTSTREAM)->GetValue("@extension").asString();
+    if (!extensions.empty())
+    {
+      std::vector<std::string> extensionsList = StringUtils::Tokenize(extensions, "|");
+      for (auto& value : extensionsList)
+      {
+        StringUtils::Trim(value);
+        if (value == filetype)
+          return true;
+      }
+    }
+  }
+
+  return false;
 }
 
 bool CInputStreamAddon::Open()
 {
-  bool ret = false;
-  if (m_addon)
-    ret = m_addon->Open(m_item);
+  m_struct.toKodi.kodiInstance = this;
+  m_struct.toKodi.free_demux_packet = cb_free_demux_packet;
+  m_struct.toKodi.allocate_demux_packet = cb_allocate_demux_packet;
+  if (!CreateInstance(&m_struct) || !m_struct.toAddon.open)
+    return false;
+
+  INPUTSTREAM props;
+  std::map<std::string, std::string> propsMap;
+  for (auto &key : m_fileItemProps)
+  {
+    if (m_item.GetProperty(key).isNull())
+      continue;
+    propsMap[key] = m_item.GetProperty(key).asString();
+  }
+
+  props.m_nCountInfoValues = 0;
+  for (auto &pair : propsMap)
+  {
+    props.m_ListItemProperties[props.m_nCountInfoValues].m_strKey = pair.first.c_str();
+    props.m_ListItemProperties[props.m_nCountInfoValues].m_strValue = pair.second.c_str();
+    props.m_nCountInfoValues++;
+  }
+
+  props.m_strURL = m_item.GetPath().c_str();
+
+  std::string libFolder = URIUtils::GetDirectory(Addon()->Path());
+  std::string profileFolder = CSpecialProtocol::TranslatePath(Addon()->Profile());
+  props.m_libFolder = libFolder.c_str();
+  props.m_profileFolder = profileFolder.c_str();
+
+  unsigned int videoWidth = 1280;
+  unsigned int videoHeight = 720;
+  if (m_player)
+    m_player->GetVideoResolution(videoWidth, videoHeight);
+  SetVideoResolution(videoWidth, videoHeight);
+
+  bool ret = m_struct.toAddon.open(&m_struct, props);
   if (ret)
   {
-    m_hasDemux = m_addon->HasDemux();
-    m_hasDisplayTime = m_addon->HasDisplayTime();
-    m_hasPosTime = m_addon->HasPosTime();
-    m_canPause = m_addon->CanPause();
-    m_canSeek = m_addon->CanSeek();
+    memset(&m_caps, 0, sizeof(m_caps));
```

Comment:
In this case not work, there is only one part defined in structure and brought errors.